ludwig/config_validation/checks.py

```diff
@@ -606,3 +607,38 @@ def check_qlora_requirements(config: "ModelConfig") -> None:  # noqa: F821
 
     if config.quantization and (not config.adapter or config.adapter.type != "lora"):
         raise ConfigValidationError("Fine-tuning and LLM with quantization requires using the 'lora' adapter")
+
+
+@register_config_check
+def check_llm_template_references(config: "ModelConfig") -> None:  # noqa: F821
+    """Checks that prompt.template includes a valid reference to a feature column."""
+    if config.model_type != MODEL_LLM:
+        return
+
+    template = config.prompt.template
+    if config.prompt.task is not None and template is not None:
+        raise ConfigValidationError("Prompt cannot have both a task and a template!")
```

Comment:
This isn't true, you are allowed to have both a task and a template. The idea is that task gets used in the template

Reply:
that was the old check